Makefile

```diff
@@ -1,8 +1,8 @@
+SRC_DIRS=src,src/sets,src/core,src/queues,src/indexed,src/maps,src/utils
 build:
-	ocamlbuild -package topkg pkg/build.native
-	./build.native build
+	rebuild -use-ocamlfind -Is $(SRC_DIRS) src/Immutable.cma
```

Comment:
As much as ocamlbuild did guarantee :)
The only thing we include as header file is `Immutable`, so it won't allow you to use `Vector` directly for example.